### ISSUE TITLE
doc: goals should be listed in order of execution

### ISSUE THREAD

**Issue by sunix** [eclipse/jkube#1527]
### Description
At the moment, in the doc https://www.eclipse.org/jkube/docs/kubernetes-maven-plugin#jkube:resource
`k8s:resource` is listed before `k8s:build`. it doesn't make sense to me as the user would try `k8s:build` and `k8s:push` first most of the time.
### Info

* Eclipse JKube version :
* Maven version (`mvn -v`) :
```

```
* Kubernetes / Red Hat OpenShift setup and version :

* If it's a bug, how to reproduce :

* If it's a feature request, what is your use case :

* Sample Reproducer Project : *[GitHub Clone URL]*

**Comment by manusa** [comment]
I really don't mind. However, note that some users might be triggering these separately or concurrently (e.g. in our Integration Tests we run these goals in parallel).

For the Outer Loop, users would usually generate the resources, helm chart, and push the chart to a shared registry. Same for the container image, users would generate and push the image to an image repository. The order only matters in case of the Inner Loop, and not for all cases.

**Comment by sunix** [comment]
Absolutely, but `resource` is very tight to `apply` and `build` to `push`, etc. It is about putting things together and simplifying navigation, searching and learning. In inner or outerloop, I don't see the difference. So it means that `k8s:helm` in 1st position and `k8s:helm-push` in the last position wouldn't be a problem?

**Comment by manusa** [comment]
As I see it this would be more or less the expected execution order:
| Inner Loop | Outer Loop |
|------------|------------|
| build      | build      |
| _push_     | push       |
| resource   | resource   |
| apply      | helm       |
| undeploy   | helm-push  |

In the outer-loop build+push and resource+helm* can be run in isolation, concurrently, or alternatively.

With regards to the current sorting, moving resource after push makes sense. But it's important to not spread the message that resource is dependent on build or push.

**Comment by sunix** [comment]
yes my point was that it doesn't make sense to have `k8s:resource`  to far from `k8s:apply` and the helm goals.
